vclib/src/commonMain/kotlin/at/asitplus/wallet/lib/data/dif/PresentationSubmissionValidator.kt

```diff
@@ -0,0 +1,89 @@
+package at.asitplus.wallet.lib.data.dif
+
+import at.asitplus.KmmResult
+import kotlinx.serialization.Serializable
+
+@Serializable
+sealed interface PresentationSubmissionValidator {
+    companion object {
+        @Throws(MissingInputDescriptorGroupException::class)
+        fun createInstance(
+            submissionRequirements: Collection<SubmissionRequirement>?,
+            inputDescriptors: Collection<InputDescriptor>,
+        ): KmmResult<PresentationSubmissionValidator> {
+            val verifier = submissionRequirements?.let { _ ->
+                SubmissionRequirementsValidator(
+                    submissionRequirements = submissionRequirements,
+                    inputDescriptorGroups = inputDescriptors.associate {
+                        it.id to (it.group ?: return KmmResult.failure(
```

Comment:
Wrap whole function in `runCatching { ... } .wrap()` and directly throw in the body. Makes it more readable imho

Reply:
fixed in commit https://github.com/a-sit-plus/kmm-vc-library/commit/65b06496293f5ec3ea9babe119596f8a137085d8